tensorrt_llm/_torch/modules/fp8_dynamic_quant_triton.py

```
@@ -0,0 +1,107 @@
+    # Match the live op `quantize_e4m3_per_tensor` EXACTLY: it multiplies by the
+    # reciprocal (amax * (1/448)), NOT divide — verified empirically (1188
+    # adversarial cases): the multiply form reproduces the op's returned scale
+    # bitwise, the divide form does not. The returned scale is used downstream as
+    # cublas_scaled_mm scale_a, so it must be bitwise-identical to baseline.
+    scale = tl.maximum(amax * INV_FP8_MAX, MIN_SF)
+    # Round to bf16 exactly like the op stores it, then widen to f32.
+    scale = scale.to(tl.bfloat16).to(tl.float32)
+    tl.store(out_ptr, scale)
+
+
+def fp8_dynamic_quant_scale(input: torch.Tensor) -> torch.Tensor:
```

Ruff (0.15.21)

[error] Function argument `input` is shadowing a Python builtin

(A002)